spec/services/aspace_client_spec.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe AspaceClient do
+  let(:url) { 'http://example.com:8089' }
+  let(:user) { 'aspace_user' }
+  let(:password) { 'aspace_password' }
+  let(:client) { described_class.new(url:, user:, password:) }
+
+  before do
+    # Authentication request
+    stub_request(:post,
+                 "#{url}/users/#{user}/login?password=#{password}").to_return(body: { session: 'token1' }.to_json)
+  end
+
+  describe '#initialize' do
+    it 'fails without the url arg' do
+      expect { described_class.new(url: nil, user: 'user', password: 'password') }.to raise_error(ArgumentError)
+    end
+
+    it 'fails without the user arg' do
+      expect do
+        described_class.new(url: 'http://example.com:8089', user: nil, password: 'password')
+      end.to raise_error(ArgumentError)
+    end
+
+    it 'fails without the password arg' do
+      expect do
+        described_class.new(url: 'http://example.com:8089', user: 'user', password: nil)
+      end.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#repositories' do
+    it 'sends an authenticated request with correct auth header to the repositories address' do
```

Comment:
can you move any "with..." from the "it" to a "context" block?

Reply:
Moved. Thank you.